module/applications/item/starting-equipment-config.mjs

```diff
@@ -18,23 +22,268 @@ export default class StartingEquipmentConfig extends DocumentSheet {
+  _canDragDrop() {
+    return this.isEditable;
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  _onDragStart(event) {
+    const entry = event.target.closest("[data-entry-id]");
+    if ( !entry ) return;
+    event.dataTransfer.setData("text/plain", JSON.stringify({
+      type: "equipment-entry", uuid: this.document.uuid, entryId: entry.dataset.entryId
+    }));
+    const box = entry.getBoundingClientRect();
+    event.dataTransfer.setDragImage(entry, box.width - 6, box.height / 2);
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  async _onDrop(event) {
+    // Try to extract the data
+    const data = TextEditor.getDragEventData(event);
+
+    // Handle re-ordering of list
+    if ( data?.entryId && (data.uuid === this.document.uuid) ) return this._onSortEntry(event, data);
+
+    // Handle dropping linked items
+    if ( data?.type !== "Item" ) return;
+    const item = await Item.implementation.fromDropData(data);
+
+    // Validate that this is a physical item
+    if ( !item.system.constructor._schemaTemplates?.includes(PhysicalItemTemplate) ) {
+      ui.notifications.error(game.i18n.format("DND5E.StartingEquipment.Warning.ItemTypeInvalid", {
+        type: game.i18n.localize(CONFIG.Item.typeLabels[item.type])
+      }));
+      return null;
+    }
+
+    // Determine where this was dropped
+    const closestDrop = event.target.closest(
+      '[data-entry-type="AND"], [data-entry-type="OR"], [data-entry-type="linked"]'
+    );
+    const { entryId, entryType } = closestDrop?.dataset ?? {};
+
+    // If no closest entry, create at top level, or if closest is a group, create inside that group
+    if ( !entryId || (entryType in EquipmentEntryData.GROUPING_TYPES) ) this.submit({ updateData: {
+      action: "add-entry",
+      entryId,
+      linkedUuid: item.uuid
+    } });
+
+    // If closest entry is linked, set its key to be this uuid
+    else if ( entryType === "linked" ) this.submit({ updateData: {
+      [`startingEquipment.${entryId}.key`]: item.uuid
+    } });
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Sort an entry on drop.
+   * @param {DragEvent} event  Triggering drop event.
+   * @param {object} data      Drag event data.
+   */
+  _onSortEntry(event, data) {
+    const dropArea = event.target.closest("[data-entry-id]");
+    const dragEntry = this.document.system.startingEquipment.find(e => e._id === data?.entryId);
+    const dropEntry = this.document.system.startingEquipment.find(e => e._id === dropArea?.dataset.entryId);
+
+    // If drag entry & drop entry are the same, or drop entry is drag entry's group, do nothing
+    if ( (dropEntry?._id === dragEntry._id) || (dropEntry?._id === dragEntry.group) ) return;
+
+    let updateData;
+    let sortBefore;
+    let target;
+
+    // If drop entry is a group, move drag entry into it
+    if ( dropEntry?.type in EquipmentEntryData.GROUPING_TYPES ) {
```

Comment:
We need to check for depth here, I was able to exceed the maximum depth when I was testing by dropping a group into another group.